tests/integration/test_multiturn.py

```diff
@@ -267,3 +270,81 @@ async def test_task_management_multi_turn_conversation():
             await task
         except asyncio.CancelledError:
             pass
+
+
+@pytest.mark.asyncio
+async def test_multi_turn_a2a_tool():
+    """Test that agents can maintain conversation context across multiple interactions."""
+
+    # Create a mock agent that simulates multi-turn conversation
+    config = AgentConfig(
+        model_id="gpt-4o-mini",  # Using real model ID but will be mocked
+        instructions=(
+            "You are a helpful assistant that remembers our conversation. "
+            "When asked about previous information, reference what was said earlier. "
+            "Keep your responses concise."
+            " If you need more information, ask the user for it."
+        ),
+        name="Structured TestResult Agent",
+        description="Agent with conversation memory for testing session management.",
+        output_type=TestResult,
+    )
+
+    agent = MockConversationAgent(config)
+
+    # Configure session management with short timeout for testing
+    serving_config = A2AServingConfig(
+        port=0,
+        task_timeout_minutes=2,  # Short timeout for testing
+    )
+
+    (task, server) = await agent.serve_async(serving_config=serving_config)
+
+    test_port = server.servers[0].sockets[0].getsockname()[1]
+    server_url = f"http://localhost:{test_port}"
+    await wait_for_server_async(server_url)
+    try:
+
+        class MainAgentAnswer(BaseModel):
```

Comment:
Although the served agent is mocked, I don't mock the main agent. I want to get an idea of how reliably the main agent calls the agent and passes it the correct taskId. If this test occasionally fails, this tells us that this initial implementation may not be robust enough.